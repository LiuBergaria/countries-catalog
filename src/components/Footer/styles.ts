import { StyleSheet } from 'react-native';

import Colors from 'src/styles/Colors';

const styles = StyleSheet.create({
  providedBy: {
    color: Colors.lightGrey,
    textAlign: 'center',
    marginVertical: 12,
    fontSize: 14,
  },
  loader: {
    marginVertical: 8,
  },
});

export default styles;
