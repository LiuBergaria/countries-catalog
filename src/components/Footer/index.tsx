import React from 'react';
import { ActivityIndicator, Linking, Text, View } from 'react-native';

import { useSafeAreaInsets } from 'react-native-safe-area-context';

import Colors from 'src/styles/Colors';

import styles from './styles';

interface IProps {
  isLoading: boolean;
}

const Footer = ({ isLoading }: IProps): JSX.Element => {
  const insets = useSafeAreaInsets();

  return (
    <View style={{ paddingBottom: insets.bottom }}>
      {isLoading && (
        <ActivityIndicator
          style={styles.loader}
          size={'large'}
          color={Colors.lightGrey}
        />
      )}

      <Text
        style={styles.providedBy}
        onPress={() => Linking.openURL('https://restcountries.com')}
      >
        Provided by https://restcountries.com/
      </Text>
    </View>
  );
};

export default React.memo(Footer);
