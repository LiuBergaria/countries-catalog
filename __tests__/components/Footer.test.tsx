import React from 'react';
import renderer from 'react-test-renderer';
import Footer from 'src/components/Footer';

jest.mock('react-native-safe-area-context', () => ({
  useSafeAreaInsets: jest.fn(() => ({ bottom: 0 })),
}));

test('renders Footer correctly', () => {
  const tree = renderer.create(<Footer isLoading={false} />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('renders Footer loading correctly', () => {
  const tree = renderer.create(<Footer isLoading={true} />).toJSON();

  expect(tree).toMatchSnapshot();
});
