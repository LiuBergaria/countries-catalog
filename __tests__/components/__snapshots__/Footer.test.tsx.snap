// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer correctly 1`] = `
<View
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#a7a7a7",
        "fontSize": 14,
        "marginVertical": 12,
        "textAlign": "center",
      }
    }
  >
    Provided by https://restcountries.com/
  </Text>
</View>
`;

exports[`renders Footer loading correctly 1`] = `
<View
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <ActivityIndicator
    color="#a7a7a7"
    size="large"
    style={
      Object {
        "marginVertical": 8,
      }
    }
  />
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#a7a7a7",
        "fontSize": 14,
        "marginVertical": 12,
        "textAlign": "center",
      }
    }
  >
    Provided by https://restcountries.com/
  </Text>
</View>
`;
